tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc

```diff
@@ -48,6 +48,54 @@ using se::DeviceMemoryBase;
 using se::dnn::AlgorithmDesc;
 using tensorflow::AutotuneResult;
 
+class ScratchAllocator : public se::ScratchAllocator {
```

Comment:
@timshen91 Any comments?

To me it seems if a "default" `se::ScratchAllocator` implementation does not exist, maybe a new one should be created in stream executor? (in a different CL?)

Reply:
FYI this comes from the original implementation of `cudnn_convlution_algorithm_picker.cc` from [here](https://github.com/tensorflow/tensorflow/blame/e7ea87f97e03360719d132a71acc1eb2f93c249f/tensorflow/compiler/xla/service/gpu/cudnn_convolution_algorithm_picker.cc). I am open to the idea of migrating the allocator to stream executor, because it does seems like there isn't a global scoped allocator owning the gpu memory somewhere.

Reply:
Oh okay. I would appreciate a PR moving it, but it is optional.